db/database.go

```diff
@@ -409,6 +416,16 @@ func NewDatabaseContext(dbName string, bucket base.Bucket, autoImport bool, opti
 		}
 		base.Infof(base.KeyAll, "Using metadata purge interval of %.2f days for tombstone compaction.", float64(context.PurgeInterval)/24)
 
+		if context.Options.AutoCompact {
+			if autoImport {
+				context.backgroundTask("Compact database", func() error { _, err := context.Compact(); return err}, time.Duration(context.PurgeInterval)*time.Hour)
+				context.backgroundTask("Compact database",
+					func() error { _, err := context.Compact(); return err},
+					time.Duration(float64(context.PurgeInterval * 60) * context.Options.AutoCompactRatio) * time.Minute)
```

Comment:
I believe I have to do this dance and swap to minutes as `time.Duration(float64(context.purgeInterval) * context.Options.AutoCompactRatio)` comes out as `0`, but there may be a nicer way...